traffic_ops/testing/api/v3/servicecategories_test.go

```diff
@@ -0,0 +1,137 @@
+	for _, sc := range testData.ServiceCategories {
+		resp, _, err := TOSession.CreateServiceCategory(sc)
+		if err != nil {
+			t.Errorf("could not CREATE service category: %v", err)
+		}
+		t.Log("Response: ", resp.Alerts)
+	}
+}
+
+func GetTestServiceCategories(t *testing.T) {
+	for _, sc := range testData.ServiceCategories {
+		resp, _, err := TOSession.GetServiceCategoryByName(sc.Name)
+		if err != nil {
+			t.Errorf("cannot GET Service Category by name: %v - %v", err, resp)
+		}
+	}
+}
+
+func UpdateTestServiceCategories(t *testing.T) {
+	firstServiceCategory := testData.ServiceCategories[0]
+	// Retrieve the Service Category by service category so we can get the id for the Update
+	resp, _, err := TOSession.GetServiceCategoryByName(firstServiceCategory.Name)
+	if err != nil {
+		t.Errorf("cannot GET Service Category by service category: %v - %v", firstServiceCategory.Name, err)
+	}
+	if len(resp) > 0 {
+		remoteServiceCategory := resp[0]
+		expectedServiceCategory := "service-category-test"
+		remoteServiceCategory.Name = expectedServiceCategory
+		var alert tc.Alerts
+		alert, _, err = TOSession.UpdateServiceCategoryByID(remoteServiceCategory.ID, remoteServiceCategory)
+		if err != nil {
+			t.Errorf("cannot UPDATE Service Category by id: %v - %v", err, alert)
+		}
+
+		// Retrieve the Service Category to check service category got updated
+		resp, _, err = TOSession.GetServiceCategoryByID(remoteServiceCategory.ID)
+		if err != nil {
+			t.Errorf("cannot GET Service Category by service category: %v - %v", firstServiceCategory.Name, err)
+		}
+
+		respServiceCategory := resp[0]
+		if respServiceCategory.Name != expectedServiceCategory {
+			t.Errorf("results do not match actual: %s, expected: %s", respServiceCategory.Name, expectedServiceCategory)
+		}
+
+		// Set the name back to the fixture value so we can delete it after
+		remoteServiceCategory.Name = firstServiceCategory.Name
+		alert, _, err = TOSession.UpdateServiceCategoryByID(remoteServiceCategory.ID, remoteServiceCategory)
+		if err != nil {
+			t.Errorf("cannot UPDATE Service Category by id: %v - %v", err, alert)
+		}
+	}
+}
+
+func ServiceCategoryTenancyTest(t *testing.T) {
+
+	toReqTimeout := time.Second * time.Duration(Config.Default.Session.TimeoutInSecs)
+	tenant4TOClient, _, err := toclient.LoginWithAgent(TOSession.URL, "tenant4user", "pa$$word", true, "to-api-v2-client-tests/tenant4user", true, toReqTimeout)
+	if err != nil {
+		t.Fatalf("failed to log in with tenant4user: %v", err.Error())
+	}
+
+	serviceCategoriesReadableByTenant4, _, err := tenant4TOClient.GetServiceCategories()
+	if err != nil {
+		t.Error("tenant4user cannot GET service categories")
+	}
+
+	// assert that tenant4user cannot read service categories outside of its tenant
+	for _, sc := range serviceCategoriesReadableByTenant4 {
+		if sc.Name == "serviceCategory1" {
+			t.Error("expected tenant4 to be unable to read service categories from tenant 3")
+		}
+	}
```

Comment:
This whole function relies on a specific structure for the test data. If I change something to do some other test, this will break even though the endpoint itself may work fine.

Instead, any specifically desired data structure should be set up and torn down by the test.